gdk4-wayland/src/wayland_device.rs

```diff
@@ -25,9 +30,15 @@ impl WaylandDevice {
     #[cfg(any(feature = "wayland_crate", feature = "dox"))]
     #[cfg_attr(feature = "dox", doc(cfg(feature = "wayland_crate")))]
     pub fn wl_keyboard(&self) -> WlKeyboard {
+        let display = self.display().downcast::<crate::WaylandDisplay>().unwrap();
         unsafe {
-            let ptr = ffi::gdk_wayland_device_get_wl_keyboard(self.to_glib_none().0);
-            Proxy::from_c_ptr(ptr as *mut wl_proxy).into()
+            let display_ptr = ffi::gdk_wayland_display_get_wl_display(display.to_glib_none().0);
+            let keyboard_ptr = ffi::gdk_wayland_device_get_wl_keyboard(self.to_glib_none().0);
```

Comment:
What policy does gtk have with returning pointers. Are these always going to return non-null pointers?

Reply:
- `gdk_wayland_display_get_wl_display` will never return null
- `gdk_wayland_device_get_wl_keyboard` can return null if the seat has no keyboard, or if the seat caps change and the keyboard gets removed

Reply:
Okay so `wl_keyboard` and friends should be wrapped in an `Option` then